src/algorithms/crypto/randomness.go

```diff
@@ -1,40 +1,41 @@
 package crypto
 
-import (
-	util "github.com/filecoin-project/specs/util"
-)
+import util "github.com/filecoin-project/specs/util"
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import abi "github.com/filecoin-project/specs/actors/abi"
 
-type Randomness = util.Randomness
+type DomainSeparationTag int
+type Randomness = abi.Randomness
 type Serialization = util.Serialization
 
-// Derive a random byte string from a domain separation tag and an arbitrary
-// serializable object.
-//
-// Note: to produce values of type Serialization, use the auto-generated method prototypes
-//   Serialize_T(T) Serialization
-// for each type T defined in .id.
-//
-// In order to derive randomness from a collection of objects, rather than just a single
-// object, define a struct at the .id level that contains those objects as member fields.
-// This will then cause a Serialize_*() method to be generated for the struct type.
-func DeriveRand(tag DomainSeparationTag, s Serialization) Randomness {
-	return _deriveRandInternal(tag, s, -1)
+const (
+	DomainSeparationTag_TicketDrawing DomainSeparationTag = 1 + iota
+	DomainSeparationTag_TicketProduction
+	DomainSeparationTag_ElectionPoStChallengeSeed
+	DomainSeparationTag_SurprisePoStChallengeSeed
+	DomainSeparationTag_SurprisePoStSelectMiners
+	DomainSeparationTag_SurprisePoStSampleSectors
+)
+
+// Derive a random byte string from a domain separation tag and the appropriate values
+func DeriveRandWithMinerAddr(tag DomainSeparationTag, tix abi.Randomness, minerAddr addr.Address) Randomness {
+	buffer := _deriveRandInternal(tag, tix, -1)
+	buffer = append(buffer, addr.Serialize_Address_Compact(minerAddr)...)
+	ret := SHA256(buffer)
+	return Randomness(ret)
 }
 
-// As in DeriveRand(), but additionally accepts an index into the implicit pseudorandom stream.
-// Index must be strictly positive.
-func DeriveRandWithIndex(tag DomainSeparationTag, s Serialization, index int) Randomness {
-	if index <= 0 {
-		panic("DeriveRandWithIndex only accepts indices > 0")
-	}
-	return _deriveRandInternal(tag, s, index)
+func DeriveRandWithEpoch(tag DomainSeparationTag, tix abi.Randomness, epoch int) Randomness {
```

Comment:
Why are we special-casing `Epoch` and `MinerAddr` as input seeds to derive randomness? The previous code allowed randomness derivation from any serializable object, with much less duplication of logic.

Reply:
Legibility. These are the only two inputs that we use to derive randomness at this point. I thought the duplication worth abstracting away the internals and making the spec easier to read in terms of what we're basing the randomness on.

Reply:
This makes the call sites easier to read. Even if they just wrapped a `DeriveRandWithCBOR`.

Reply:
yeah they more or less do just that, wrapping a `_deriveRandInternal` call

Reply:
And the day we want to add another source of entropy we can just add a new wrapper